include/zephyr/device.h

```diff
@@ -319,7 +319,7 @@ typedef int16_t device_handle_t;
  * @param compat lowercase-and-underscores devicetree compatible
  * @return a pointer to a device
  */
-#define DEVICE_DT_GET_ONE(compat)					    \
+#define DEVICE_DT_GET_ANY(compat)					    \
```

Comment:
No way. You are in effect subtly changing the semantics of an existing API. This is evil; I can't countenance this!

Reply:
> No way. You are in effect subtly changing the semantics of an existing API. This is evil; I can't countenance this!

Btw, `DEVICE_DT_GET_ANY` and `DEVICE_DT_GET_ONE` end up returning the same, only that `DEVICE_DT_GET_ANY` returns `NULL` if none are found. So to me, current semantics are also wrong.

"any device":
any, adv.: At least one; of at least one kind (**NOT true in our current API, as it can return NULL**) 

"one device":
one, det: used for referring to a single person or thing when there are others of the same type

So adjusting to `ONE`/`ONE_OR_NULL` makes sense, IMO.

Reply:
>No way. You are in effect subtly changing the semantics of an existing API.

How so? Unless the underlying implementation is modified I cannot see how `GET_ONE` differs from `GET_ANY` semantically without additional context.

Reply:
The thing is that if we only `s/DEVICE_DT_GET_ANY/DEVICE_DT_GET_ONE_OR_NULL`, we leave `ONE` users as-is, we can easily move from `ANY` to `ONE_OR_NULL` and deprecate `ANY`. If renaming `ANY` to `ONE`, we are changing the behavior of an existing API (it no longer returns NULL, but throws a compiler error, so we'll likely break users who intentionally wanted an optional device).